src/main/cpp/src/cast_long_to_binary_string.cu

```diff
@@ -30,35 +30,44 @@ namespace spark_rapids_jni {
 
 namespace detail {
 namespace {
-struct long_to_binary_string_fn {
-  cudf::column_device_view d_longs;
-  cudf::size_type* d_sizes;
-  char* d_chars;
-  cudf::detail::input_offsetalator d_offsets;
-
-  __device__ void long_to_binary_string(cudf::size_type idx)
-  {
-    auto const value = d_longs.element<int64_t>(idx);
-    char* d_buffer   = d_chars + d_offsets[idx];
-    for (auto i = d_sizes[idx] - 1; i >= 0; --i) {
-      *d_buffer++ = '0' + ((value & (1UL << i)) >> i);
-    }
+CUDF_KERNEL void compute_output_size_kernel(cudf::column_device_view d_longs,
+                                            cudf::size_type* d_sizes)
+{
+  auto const tid = cudf::detail::grid_1d::global_thread_id();
+  if (tid >= d_longs.size()) { return; }
+  if (d_longs.is_null(tid)) {
+    d_sizes[tid] = 0;
+    return;
   }
+  // If the value is 0, the size should be 1
+  d_sizes[tid] = max(1, 64 - __clzll(d_longs.element<int64_t>(tid)));
+}
 
-  __device__ void operator()(cudf::size_type idx)
-  {
-    if (d_longs.is_null(idx)) {
-      if (d_chars == nullptr) { d_sizes[idx] = 0; }
-      return;
-    }
-    if (d_chars != nullptr) {
-      long_to_binary_string(idx);
-    } else {
-      // If the value is 0, the size should be 1
-      d_sizes[idx] = max(1, 64 - __clzll(d_longs.element<int64_t>(idx)));
-    }
+CUDF_KERNEL void long_to_binary_string_kernel(cudf::column_device_view d_longs,
```

Comment:
num_threads_per_row is a const value, we can make num_threads_per_row as a template parameter.